client/src/app/pages/controls/stakeholder-groups/components/stakeholder-group-form.tsx

```diff
@@ -121,36 +170,22 @@ export const StakeholderGroupForm: React.FC<StakeholderGroupFormProps> = ({
         };
       });
 
-    const payload: StakeholderGroup = {
+    const payload: Omit<StakeholderGroup, "id"> = {
```

Comment:
Why is this necessary?

Reply:
Because Id is always present but during creation.
This avoids to test for undefined Id when there is an ID.
I've updated description for more details

Reply:
Thats a good idea.

Reply:
Thanks.
I didn't mean to make so many changes but once I touched the ID it was a long journey to fix everything around it.
There are still some resources using optional ID but those weren't impacted so we can leave them as tech debt.